api4/handlers.go

```diff
@@ -8,14 +8,16 @@ import (
 
 	"github.com/mattermost/gziphandler"
 
+	"github.com/mattermost/mattermost-server/v6/model"
 	"github.com/mattermost/mattermost-server/v6/web"
 )
 
 type Context = web.Context
 
 // APIHandler provides a handler for API endpoints which do not require the user to be logged in order for access to be
 // granted.
-func (api *API) APIHandler(h func(*Context, http.ResponseWriter, *http.Request)) http.Handler {
+func (api *API) APIHandler(h func(*Context, http.ResponseWriter, *http.Request), scopes ...model.Scope) http.Handler {
+	handlerScopes := scopeFromArgs(scopes)
```

Comment:
See my comment in `scope.go`; 1/5 with my proposed change you just do `Scopes(scopes)`, no need for different representations. You can still use the custom `Scopes` type for all the method bindings, but no need to change the representation?

Reply:
If we go with Scopes(scopes), we would be changing the whole meaning to:
- Empty = deny
- Includes "" = allow
- Anything else = scoped

We could go that direction, but not sure what is more confusing as a whole, doing this big change here, or keep it as is.